### PR TITLE
[Triton-Gluon-MLA-GFX950] return_lse: full decode + merged fp32 lse

### PR DESCRIPTION
**Pull request by Dewei-Wang-sh**
return_lse now runs the full decode (stage-1 + stage-2 reduce) and returns the merged fp32 lse [batch, nhead] (e_max + log(e_sum) over the whole sequence), supported for all regimes.

- per-split lse kept in a separate fp32 mid_lse buffer (not packed into the bf16 O), fixing the bf16 precision loss in the stage-2 merge
- stage-2 (_mla_softmax_reducev_kernel) writes final_lse; NUM_KV_SPLITS==1 fast path writes it directly in stage-1
- test_mla.py: validate returned lse vs lse_ref; README updated

### CODE REVIEW

File: aiter/ops/triton/gluon/mla_decode_gluon.py [hunk 1]
```diff
@@ -732,18 +731,20 @@ def mla_decode_gluon(
...
-        workload where the KV cache is sharded across GPUs and the host does
-        the cross-GPU softmax merge. Only the bh16bn64 regime (nhead<=16, bf16
-        KV) supports it. The stage-2 reduce is skipped; the caller's `o` is the
-        per-split logits buffer [batch, nhead, NUM_KV_SPLITS, kv_lora_rank]
-        (bf16), and a per-split fp32 lse [batch, nhead, NUM_KV_SPLITS] is
-        allocated here and returned: (o, lse). Tuned for batch*ctx_len in the
-        10K..100K total-token range; NUM_KV_SPLITS = max(1, 256 // batch_size).
+    Always runs the full decode (stage-1 + stage-2 reduce, or the stage-1-only
+    fast path when NUM_KV_SPLITS==1) and writes the final attention into the
+    caller's `o` ([batch, nhead, kv_lora_rank]).
```
[the remark this answers is not in the document]
Comment (Dewei-Wang-sh) [reply aiter/ops/triton/gluon/mla_decode_gluon.py:729]: fixed

File: op_tests/test_mla.py [hunk 2]
```diff
@@ -526,14 +526,11 @@ def test_absorb_decode_gluon():
     def test_absorb_decode_gluon_bh16(name):
         # Shared bh16bn{64,128} runner. The wrapper dispatches on
         # (nhead, kv dtype): name='bh16bn128' -> cast kv to fp8;
-        # name='bh16bn64' -> keep bf16. When the outer -lse/--return_lse flag is
-        # set, run the stage-1-only (DCP) path: per-GPU stage-1, no intra-GPU
-        # reducev. Host-side LSE-merge over splits validates equivalence to the
-        # unified attention reference.
+        # name='bh16bn64' -> keep bf16. Always runs the full decode (stage-1 +
+        # stage-2 reduce). With -lse/--return_lse the kernel also returns the
+        # merged fp32 lse [batch, nhead], validated against lse_ref.
```
[the remark this answers is not in the document]
Comment (Dewei-Wang-sh) [reply op_tests/test_mla.py:531]: not aim to support

**Review by valarLip — APPROVED** [review]
LGTM
